src/olympia/search/filters.py

```diff
@@ -335,6 +336,32 @@ class AddonRecommendedQueryParam(AddonQueryParam):
     es_field = 'is_recommended'
 
 
+class AddonPromotedQueryParam(AddonQueryParam):
+    query_param = 'promoted'
+    reverse_dict = {
+        group.api_name: id_
+        for (id_, group) in ENABLED_PROMOTED_GROUPS_BY_ID.items()}
+    valid_values = ENABLED_PROMOTED_GROUPS_BY_ID.keys()
+
+    def get_app(self):
+        return (
+            AddonAppQueryParam(self.request).get_value()
+            if AddonAppQueryParam.query_param in self.request.GET
+            else None)
+
+    def get_es_query(self):
+        query = [Q(
+            self.operator,
+            **{'promoted.group_id': self.get_value()})]
+
+        if app := self.get_app():
+            query.append(
+                Q(self.operator, **{'promoted.application_id': app}) |
+                ~Q('exists', field='promoted.application_id'))
```

Comment:
I assume this is here for a good reason, but why ?

Reply:
When application_id is `None` it seems to just not get stored (unless I'm missing something)

Reply:
Ah, yes. Deserves a comment I think.

Reply:
https://github.com/mozilla/addons-server/commit/23a5ef9c6a2bf61a6714df48ab406d13666233b7